### DISCUSSION TITLE
Perception warning in unexpected place

### DISCUSSION THREAD

**Discussion by bunnyhero** [pointfreeco/swift-perception#35] [Q&A]
I'm not sure if this is an issue or if it's working as expected.

This code, using TCA 1.8, and targeting iOS 16 + 17, results in the "Perceptible state was accessed but is not being tracked" runtime warning, even though the single access to the bindable `store` is wrapped in `WithPerceptionTracking`. I can make the warning go away if I throw an additional `WithPerceptionTracking` inside the `GeometryReader`, even though the GeometryReader's contents have no references to anything Perceptible.

I can also make the warning go away if I change the order of the `TextField` and the `GeometryUser()`.

Is this a false alarm? Thanks for any insight.

```Swift
import SwiftUI
import ComposableArchitecture

@Reducer
struct MyFeature {
    @ObservableState
    struct State {
        var message: String = ""
    }

    enum Action: BindableAction {
        case binding(BindingAction<State>)
    }

    var body: some ReducerOf<Self> {
        BindingReducer()

        Reduce { state, action in
            return .none
        }
    }
}

struct MyView: View {
    @Perception.Bindable var store: StoreOf<MyFeature>
    var body: some View {
        WithPerceptionTracking {
            VStack {
                GeometryUser()
                TextField("Message", text: $store.message)
            }
        }
    }
}


struct GeometryUser: View {
    var body: some View {
        GeometryReader { geo in
            // If we put WithPerceptionTracking around the Text(), the
            // warning goes away.
            Text("Text")
                .tag(geo.size.width)
                // ^ This is contrived, but it's a minimal example of an
                // identity change when the geometry changes. The actual
                // code where we encountered this issue makes more sense.
        }
    }
}
```

**Comment by mbrandonw** [comment]
Hi @bunnyhero, when I paste this code into a project I do not see any perception warnings. Can you provide a minimal project that demonstrates the behavior you are seeing?

**Reply by bunnyhero** [reply]
Sure thing! Hope it's OK to attach zips-- if not I'll make a GitHub repo. And thanks for taking a look
[perceptionissue.zip](https://github.com/pointfreeco/swift-perception/files/14312587/perceptionissue.zip)
